vmm/src/vmm_config/mod.rs

```diff
@@ -16,3 +19,62 @@ pub mod net;
 #[cfg(feature = "vsock")]
 /// Wrapper for configuring the vsock devices attached to the microVM.
 pub mod vsock;
+
+// TODO: Migrate the VMM public-facing code (i.e. interface) to use stateless structures,
+// for receiving data/args, such as the below `RateLimiterConfig` and `TokenBucketConfig`.
+// Also todo: find a better suffix than `Config`; it should illustrate the static nature
+// of the enclosed data.
+// Currently, data is passed around using live/stateful objects. Switching to static/stateless
+// objects will simplify both the ownership model and serialization.
+// Public access would then be more tightly regulated via `VmmAction`s, consisting of tuples like
+// (entry-point-into-VMM-logic, stateless-args-structure).
+
+/// A public-facing, stateless structure, holding all the data we need to create a TokenBucket
+/// (live) object.
+#[derive(Clone, Copy, Debug, Default, Deserialize)]
+pub struct TokenBucketConfig {
+    /// See TokenBucket::size.
+    pub size: u64,
+    /// See TokenBucket::one_time_burst.
+    pub one_time_burst: Option<u64>,
+    /// See TokenBucket::refill_time.
+    pub refill_time: u64,
+}
+
+impl TokenBucketConfig {
+    /// Convert the stateless `self` into a live `TokenBucket` object.
+    ///
+    /// This would look nicer if we were to implement `Into<TokenBucket>` (or, more generally,
```

Comment:
This comment is more like a TODO, so please move it inside the function and do not use doc comments.

Reply:
Done.